### ISSUE TITLE
Bug in code formatter in REPL

### ISSUE THREAD

**Issue by twolodzko** [uiua-lang/uiua#180]
The code formatter in REPL removes some characters from the code:

![image](https://github.com/uiua-lang/uiua/assets/3750140/89aea059-fe26-4355-9b6a-d3d66fdd522f)

This seems not to be a formatting bug per se, if you add the following test to the format.rs file, it passes.

```rust
#[cfg(test)]
mod tests {
    use super::FormatConfig;

    #[test]
    fn format_str() {
        let config = FormatConfig::default();
        assert_eq!(
            super::format_str("(+1|*2) 1 5", &config).unwrap().output,
            "(+1|×2) 1 5\n"
        );
    }
}
```

Looking at your refactor of the REPL code, I don't see any obvious culprit, but would worry that it could be something in the `lsp::spans` function, so it could be something more serious than pure formatting. But I didn't dig deeper in your code yet, so this is a pure guess.

**Comment by kaikalii** [comment]
Should be fixed in c62766ca95dfe69fd03f.
There may be some spans I've missed though.

**Comment by kaikalii** [comment]
I fixed some more bugs related to this in 0c6f2c090c9c5 and d5b4502acf7a1749b.
